receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -370,12 +382,75 @@ func (p *StatsDParser) Aggregate(line string, addr net.Addr) error {
 			// No action.
 		}
 	}
+}
 
-	return nil
+// statsDMetricIter allows alloc-free iteration over statsDMetric values.
+// This is helpful to support multiple values in a single statsd metric 
+// such as timers, etc.
+type statsDMetricIter struct {
+	statsDMetricProperties
+
+	// valueStr contains potentially many values separated by ":".
+	valueStr string
+	valueCount int
 }
 
-func parseMessageToMetric(line string, enableMetricType bool, enableSimpleTags bool) (statsDMetric, error) {
-	result := statsDMetric{}
+type statsDMetricProperties struct {
+	description statsDMetricDescription
+	addition    bool
+	unit        string
+	sampleRate  float64
+	timestamp   uint64
+}
+
+func (i statsDMetricIter) len() int {
```

Comment:
Should use pointer receivers here, it won't cause allocations or anything. Since this is likely not getting inlined, it's copying receiver struct by value each call.
```suggestion
func (i *statsDMetricIter) len() int {
```